p2p/pex_reactor_test.go

```diff
@@ -19,7 +19,11 @@ func TestPEXReactorBasic(t *testing.T) {
 
 	dir, err := ioutil.TempDir("", "pex_reactor")
 	require.Nil(err)
-	defer os.RemoveAll(dir)
+	defer func() {
```

Comment:
lets make a function for it

Reply:
dug a little deeper. These types of defer is the example metalinter uses for `// nolint: errcheck` and it's idiomatic go anyways - I probably went a little too crazy on the defer funcs ...

Reply:
excellent news!